pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/service_proxy_transport.go

```diff
@@ -0,0 +1,237 @@
+package openshiftkubeapiserver
+
+import (
+	"fmt"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/api/core/v1"
+	kapierrors "k8s.io/apimachinery/pkg/api/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	listersv1 "k8s.io/client-go/listers/core/v1"
+	restclient "k8s.io/client-go/rest"
+	cache "k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+// serviceCABundleRoundTripper creates a new RoundTripper (or uses cachedTransport) for each request with serverName and
+// caBundle input into the TLSClientConfig. It is expected that caBundle is kept up to date and cachedTransport
+// cleared by the serviceCABundleUpdater controller.
+type serviceCABundleRoundTripper struct {
+	serverName      string
+	caBundle        atomic.Value
+	cachedTransport http.RoundTripper
+}
+
+func (r *serviceCABundleRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	rt := r.cachedTransport
+	if rt == nil {
```

Comment:
this smells racy.  I could end up with different cachedTransports in different threads for a short while, right?